tests/normalize/test_model_item_normalizer.py

```diff
@@ -0,0 +1,356 @@
+    extractor.close_writers(load_id)
+    extractor.commit_new_load_package(load_id, schema)
+    return load_id
+
+
+def create_schema_with_complete_columns(
+    table_name: str, data_type: str, columns: List[str]
+) -> Schema:
+    cols = [utils.new_column(column_name=col, data_type=data_type) for col in columns]  # type: ignore[arg-type]
+    table = utils.new_table(table_name, columns=cols)
+    schema = Schema("my_schema")
+    schema.update_table(table)
+    return schema
+
+
+def extract_normalize_retrieve(
+    model_normalize: Normalize,
+    model: SqlModel,
+    schema: Schema,
+    table_name: str,
+    fallback_dialect: str,
+) -> Tuple[str, str, str]:
+    # Extract and normalize the model
+    load_id = extract_model(model_normalize.normalize_storage, model, schema, table_name)
+    with ThreadPoolExecutor(max_workers=1) as pool:
+        model_normalize.run(pool)
+
+    # Retrieve the normalized query from the storage
+    model_files = model_normalize.load_storage.list_new_jobs(load_id)
+    model_full_path = model_normalize.load_storage.normalized_packages.storage.make_full_path(
+        model_files[0]
+    )
+    with FileStorage.open_zipsafe_ro(model_full_path, "r", encoding="utf-8") as f:
+        select_dialect, normalized_select_query = read_dialect_and_sql(
+            file_obj=f,
+            fallback_dialect=fallback_dialect,
+        )
+
+    return select_dialect, normalized_select_query, load_id
+
+
+@pytest.mark.parametrize("caps", MODEL_CAPS, indirect=True, ids=DESTINATIONS_SUPPORTING_MODEL)
+def test_simple_model_normalizing(
+    caps: DestinationCapabilitiesContext, model_normalize: Normalize
+) -> None:
+    """
+    This test demonstrates how a model with simple sql query is transformed by the normalizer for each destination capabilities.
+    Explicit queries are used to clearly show the expected transformation process.
+    """
+    # TODO: tests for dialects teradata, mysql, presto and athena
+    dialect = caps.sqlglot_dialect
+    model = SqlModel.from_query_string(query="SELECT a, b FROM my_table", dialect=dialect)
+
+    # Ensure the schema contains the table "my_table" with columns a, b
+    schema = create_schema_with_complete_columns("my_table", "text", ["a", "b"])
+
+    select_dialect, normalized_select_query, load_id = extract_normalize_retrieve(
+        model_normalize, model, schema, "my_table", dialect
+    )
+
+    assert select_dialect == dialect
+    if dialect == "duckdb":
+        assert (
+            f"SELECT subquery.a AS a, subquery.b AS b, '{load_id}' AS _dlt_load_id, UUID() AS"
+            " _dlt_id FROM (SELECT a, b FROM my_table) AS subquery\n"
+            == normalized_select_query
+        )
+    elif dialect == "bigquery":
+        assert (
+            f"SELECT subquery.a AS a, subquery.b AS b, '{load_id}' AS _dlt_load_id, GENERATE_UUID()"
+            " AS _dlt_id FROM (SELECT a, b FROM my_table) AS subquery\n"
```

Comment:
maybe we should use a different identifier than "subquery" in the normalizer, something more abstract so that we do not have collissions with anything the user may provide? maybe "_dlt_subquery" or something like this?